### DISCUSSION TITLE
Unable to build - Error executing /usr/bin/ldc2: Killed - SOLVED

### DISCUSSION THREAD

**Discussion by justinyunke** [abraunegg/onedrive#2312] [Q&A] [answered]
The current version of onedrive in the Debian apt repo is:
onedrive/stable,now 2.4.10-1 amd64 [residual-config]

...and I was hit by the "Conversion positive overflow" bug that was fixed last year. So, I tried compiling:
```
Error:
ldmd2 -w -g -O -J. -L-lcurl -L-lsqlite3  -L-ldl src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedrive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d src/progress.d src/arsd/cgi.d -ofonedrive
Error: Error executing /usr/bin/ldc2: Killed
make: *** [Makefile:102: onedrive] Error 1
```
Environment:
Latest version (2.4.23)
Debian 11.6
Linux slmo-util-1 5.10.0-21-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux

I had to manually install a lot of libraries to get this far, perhaps I picked the wrong one:
```
apt install ldc
apt install pkg-config
apt install libcurl4
apt install curl
apt install libcurl4-openssl-dev
apt install libsqlite3-dev
```
Anyone have any tips?

Thank you!

P.S. I just now found the instructions about using the OpenSUSE repo.  I'll give that a shot if someone doesn't have a tip on compiling.

**Comment by abraunegg** [comment, marked as answer]
@justinyunke 
> Anyone have any tips?

Do not use the Debian repo's for this client. As per the install document:

![image](https://user-images.githubusercontent.com/4956234/221004330-b577dc69-647e-4f60-b191-909e1de6a11d.png)

If you are fixated on building the client for Debian, please follow the correct installation process as per https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#dependencies-ubuntu-18x---ubuntu-22x--debian-9---debian-11---x86_64
```
sudo apt install build-essential
sudo apt install libcurl4-openssl-dev libsqlite3-dev pkg-config git curl
curl -fsS https://dlang.org/install.sh | bash -s dmd
```

**Reply by justinyunke** [reply]
Thanks for reaching out!  The issue was lack of memory.

Stumbled upon this in dmesg: "Out of memory: Killed process 17453 (ldc2) total-vm:1376644kB, anon-rss:769780kB, file-rss:0kB, shmem-rss:0kB, UID:1001 pgtables:2672kB oom_score_adj:0" - the build instance I was using (VMware) only had 1GB of RAM and 512MB of swap.  I bumped it to 2GB of ram, installed build-essential for cc (thank you!) and I'm running great, now.
